compiler/damlc/daml-doc/src/DA/Daml/Doc/Extract/Templates.hs

```diff
@@ -297,22 +297,26 @@ unControllerExpr e
   , Just (body, [_, sndPat]) <- unMatchGroup matchGroup
   = case sndPat of
       WildPat _ -> Nothing -- If `y` is _, we're in an Archive.
-      _ -> unLets body
+      _ -> unLets1 body
 
   -- \x -> bypassReduceLambda $ \y -> Body
   | HsLam _ matchGroup <- e
   , Just (body, [_]) <- unMatchGroup matchGroup
   , Just innerLambda <- stripDesugarApplication "bypassReduceLambda" body
   , HsLam _ matchGroup <- innerLambda
   , Just (innerBody, _) <- unMatchGroup matchGroup
-  = unLets innerBody
+  = unLets1 innerBody
 
   | otherwise = Nothing
   where
-    -- let _ = this in let _ = arg in ...
-    unLets :: HsExpr GhcPs -> Maybe (HsExpr GhcPs)
-    unLets (HsLet _ _ (L _ (HsLet _ _ (L _ body)))) = Just body
-    unLets _ = Nothing
+    -- Drop as many let binds as possible, at least one (will be stopped by the application of toParty or concat)
+    -- We cannot drop a fixed number of binds, as the number of binds depends on the let body of the template
+    unLets1 :: HsExpr GhcPs -> Maybe (HsExpr GhcPs)
+    unLets1 (HsLet _ _ (L _ body)) = Just $ unLets body
+    unLets1 _ = Nothing
+    unLets :: HsExpr GhcPs -> HsExpr GhcPs
+    unLets (HsLet _ _ (L _ body)) = unLets body
+    unLets body = body
```

Comment:
I'm not sure if this will work when the controller expression itself contains a let binding, but I don't think it's worth the effort to fix right now - especially if we're dropping template-level let blocks soon 🤞

Reply:
Just checked, it works as expected, since the `toParties` is wrapped around the let binding, so a controller of `let x = p in x` becomes `let _ = blah in let _ = blah in Desugar.toParties (let x = p in x)`